### ISSUE TITLE
Long title names in Tray

### ISSUE THREAD

**Issue by varmanishant** [joewing/jwm#316]
Can the long title names in tray be ended like [ Firefox - Recent Developm... ] spaced equally on the left and right side? Also is it possible to have borders for inactive windows?

**Comment by joewing** [comment]
Making long title names cut off in a more natural way is something that would be nice.  I think it should probably end on full-letter boundaries and possibly insert ellipses.  This would be useful for both window titles and tray buttons I think.  Minimized windows pose an additional complication in that they are placed in square brackets, so it would be nice to always have the closing bracket.

I'm not sure what you mean by borders for inactive windows.

Although JWM could display something different for the window title (anything available via `xprop` is fair game), I tend to think that JWM shouldn't since that's the intended use of the title property.  Of course, I think a case could be made for using `_NET_WM_ICON_NAME` i(or `WM_ICON_NAME`) nstead of `_NET_WM_NAME` (or `WM_NAME`) for the task bar.  Unfortunately, I think those are often set to the same value (they appear to be the same for Firefox at least), making them fairly useless.  In any case, that's more an issue with the application than JWM.  I get that JWM needs to be able to work around bad application behavior, but I really would like to avoid doing so.

**Comment by technosaurus** [comment]
I did some experimenting with the group option to do only icons and put the normal label as the tooltip.  I really like it, but my implementation was kind of buggy, so l never did a pull request.  It fits better with my tendency to keep many windows open ...often many different applications.  I usually have a few different browsers open, a text editor, image editor/viewer, a spreadsheet and a terminal - just for everyday usage.  That pretty much fills the tray and any out of the normal application causes everything to squish so much I cannot read the text anyhow.  It's maybe the only thing windows actually does partially right.

**Comment by varmanishant** [comment]
This is from tint2 https://i.imgsafe.org/bd59172b87.png. Its bolded (by me) and equi-spaced (by tint2). That is also decent. Btw, I can distinguish the tabs which is what I meant by having border for inactive tabs. Each tabs has a slightly different shade background and a seperator. In JWM currently is there a way to disntinguish inactive tabs from the Tray itself? Didn't see an option for that.

I think icons only is also nice, but I think it should be configurable. From Windows 7, I guess they only have icons and no text's.

**Comment by joewing** [comment]
As far as inactive items in the task list are concerned, you can have a different style for those than the rest of the tray.  For example:
```xml
<TrayStyle>
    <Background>green</Background>
</TrayStyle>
<TaskListStyle>
    <Background>blue</Background>
    <Active>
        <Background>red</Background>
    </Active>
</TaskListStyle>
```
That would give the tray a green background, inactive tasks blue backgrounds, and the active task a red background.  Is that what you mean?

**Comment by varmanishant** [comment]
Thanks, yes this is what I meant. Sorry for the delay in my response.

**Comment by yetanothergeek** [comment]
Ending on letter boundaries would be nice, but I'm not so sure about ellipsis. Given a choice between two more letters or three dots, I would rather see "Trumpet" than "Trump..."

**Comment by varmanishant** [comment]
Yeah but the problem is you wont see a difference between "Trumphet" and "Trumphet For Sale", both would look "Trumphet" which is a bit confusing.

**Comment by int-72h** [comment]
Looking at tint2 for its ... feature, [this](https://gitlab.com/o9000/tint2/blob/master/src/taskbar/taskbarname.c) appears to be where it's calculated. Seems like a simple check, but I guess it's based on font size as well.
